tfhe/benches/integer/glwe_packing_compression.rs

```diff
@@ -41,23 +45,85 @@ fn cpu_glwe_packing(c: &mut Criterion) {
         assert_eq!(bit_size % log_message_modulus, 0);
         let num_blocks = bit_size / log_message_modulus;
 
-        let ct = cks.encrypt_radix(0_u32, num_blocks);
+        let bench_id_pack;
+        let bench_id_unpack;
 
-        let mut builder = CompressedCiphertextListBuilder::new();
+        match BENCH_TYPE.get().unwrap() {
+            BenchmarkType::Latency => {
+                let ct = cks.encrypt_radix(0_u32, num_blocks);
 
-        builder.push(ct);
+                let mut builder = CompressedCiphertextListBuilder::new();
+
+                builder.push(ct);
+
+                bench_id_pack = format!("{bench_name}::pack_u{bit_size}");
+                bench_group.bench_function(&bench_id_pack, |b| {
+                    b.iter(|| {
+                        let compressed = builder.build(&compression_key);
+
+                        _ = black_box(compressed);
+                    })
+                });
 
-        let bench_id = format!("{bench_name}::pack_u{bit_size}");
-        bench_group.bench_function(&bench_id, |b| {
-            b.iter(|| {
                 let compressed = builder.build(&compression_key);
 
-                _ = black_box(compressed);
-            })
-        });
+                bench_id_unpack = format!("{bench_name}::unpack_u{bit_size}");
+                bench_group.bench_function(&bench_id_unpack, |b| {
+                    b.iter(|| {
+                        let unpacked: RadixCiphertext =
+                            compressed.get(0, &decompression_key).unwrap().unwrap();
+
+                        _ = black_box(unpacked);
+                    })
+                });
+            }
+            BenchmarkType::Throughput => {
+                let elements = throughput_num_threads();
+                // FIXME thread usage seemed to be somewhat more "efficient".
+                //  For example, with bit_size = 2, my laptop is only using around 2/3 of the
+                // available threads  Thread usage increases with bit_size = 8 but
+                // still isn't fully loaded.
+                bench_group.throughput(Throughput::Elements(elements));
+
+                let builders = (0..elements)
+                    .map(|_| {
+                        let ct = cks.encrypt_radix(0_u32, num_blocks);
+                        let mut builder = CompressedCiphertextListBuilder::new();
+                        builder.push(ct);
+
+                        builder
+                    })
+                    .collect::<Vec<_>>();
```

Comment:
this won't fill a compressed cipehrtext list, we have 512 bits of capacity per builder before we are creating several lists

also begs the question are we interested in one builder with a lot of ciphertexts or several builders (like it's currently written) ?

Reply:
Do we care about filling them ? We just want to perform the same kind of operation in parallel. We don't want to fill-up the list but fully loading the CPU first.

Reply:
actually that's fine as the throughput will be tested for different levels of fullness of the list